### PR TITLE
48 initialize wallet service

### LINKED ISSUE #5: Implement wallet-api endpoints
**Description:**

- Implement the required wallet endpoints for Sprint 2, which includes initializing a wallet and getting the balance of a wallet.
- Write unit tests and integration tests for all service functions/endpoints.

**Acceptance Criteria**

- [x]  The middleware/service can retrieve the amount from the Blockchain and pass it on to the UI, given a walletID
- [x] The middleware/service can initialize a new wallet for a user, given a walletID

### LINKED ISSUE #48: Initialize the wallet service
**Description:**

- Create a docker container for the service
- Initialize required endpoints for Sprint 2, including the ability to initialize a new wallet and retrieve a wallet's balance.

**Acceptance Criteria:**
- [x] The frontend app is able to make HTTP requests to the service to retrieve data as described above.

### PR DESCRIPTION
**Pull request by akshaysharma21**
Initialized the wallet service.

how to test:

local: navigate to the wallet service and run "python -m pytest" to run the test.
docker: spin up the containers using "docker-compose up" and go to "localhost/wallet". this should show you a welcome message.

Note: The endpoints won't work on docker containers as they depend on the blockchain endpoints

### CODE REVIEW

**Review by abhisachdev17 — APPROVED** [review]
LGTM

File: backend/services/wallet/src/routes.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from src import app
+
+@app.route("/")
+def index():
+    return "Hello from your wallet"
```
Comment (ZackHolmberg) [inline backend/services/wallet/src/routes.py:16]: We need endpoints for adding a new wallet and getting a wallet's amount, as well as adding/subtracting from a wallet. All will access the wallet through a walletId passed in the request. Please refer to the Blockchain endpoints, which these endpoints will simply be hitting.

File: backend/services/wallet/tests/functional/test_routes.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    test_client = app.test_client()
+
+    ctx = app.app_context()
+    ctx.push()
+
+    yield test_client
+
+    ctx.pop()
+
+def test_home_page(test_client):
```
Comment (ZackHolmberg) [inline backend/services/wallet/tests/functional/test_routes.py:25]: Will need new tests for the new endpoints described in the previous comment.

**Review by ZackHolmberg — APPROVED** [review]

**Review by ZackHolmberg — CHANGES_REQUESTED** [review]
I meant to request changes and not approve. Please see above comments

File: backend/services/wallet/src/routes.py (hunk 1 above)
Comment (akshaysharma21) [reply backend/services/wallet/src/routes.py:16]: done

File: backend/services/wallet/tests/functional/test_routes.py (hunk 2 above)
Comment (akshaysharma21) [reply backend/services/wallet/tests/functional/test_routes.py:25]: done

File: backend/services/wallet/endpoints.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+Registers a wallet for a new user in the blockchain and returns the server message.
+
+**Example return:**
+
+```json
+201 
+{
+  "success": boolean
```
Comment (ZackHolmberg) [inline backend/services/wallet/endpoints.md:22]: I think here you can just use `true` as I don't think `false` will ever appear in the success path, so we can be explicit

File: backend/services/wallet/src/routes.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import json
+
+if os.environ.get('SERVICE_IN_DOCKER',False):             
+    sys.path.append(os.path.abspath(os.path.join('..', '')))
+else:
+    sys.path.append(os.path.abspath(os.path.join('../..', '')))
+
+from shared import HttpCode, FailureReturnString
+
+@app.route("/")
```
Comment (ZackHolmberg) [inline backend/services/wallet/src/routes.py:14]: I think we could just remove this endpoint and its corresponding test, we don't really need it

File: backend/services/wallet/tests/functional/test_routes.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
```
Comment (ZackHolmberg) [inline backend/services/wallet/tests/functional/test_routes.py:25]: Remove this if you remove the endpoint

**Review by ZackHolmberg — APPROVED** [review]

File: backend/services/wallet/src/routes.py (hunk 4 above)
Comment (akshaysharma21) [reply backend/services/wallet/src/routes.py:14]: Do you mind if we keep these home endpoints until we have everything ready to ship? They are quite useful to check if the service is up while in dev.

File: backend/services/wallet/tests/functional/test_routes.py (hunk 5 above)
Comment (akshaysharma21) [reply backend/services/wallet/tests/functional/test_routes.py:25]: will do.

File: backend/services/wallet/src/routes.py (hunk 4 above)
Comment (ZackHolmberg) [reply backend/services/wallet/src/routes.py:14]: Yep fair enough, sounds good!

**Pull request merged**